src/libmongoc/tests/test-mongoc-bulk.c

```diff
@@ -3548,7 +3575,20 @@ test_bulk_split (void)
    BSON_ASSERT (!r);
 
    /* all 100,010 docs were inserted, either by the first or second bulk op */
-   ASSERT_COUNT (n_docs, collection);
+   {
+      bson_t count_opts = BSON_INITIALIZER;
+      ASSERT_OR_PRINT (
+         mongoc_client_session_append (session, &count_opts, &error), error);
+      int64_t count = mongoc_collection_count_documents (collection,
```

Comment:
Confirming, is this the only test using `ASSERT_COUNT` that must be updated to ensure causal consistency because it is the only(?) test that whose assertion depends on multiple bulk write operations?

Reply:
> Confirming, is this the only test using `ASSERT_COUNT` that must be updated to ensure causal consistency because it is the only(?) test that whose assertion depends on multiple bulk write operations?

I expect many tests implicity assume "read-your-own-writes" semantics without guaranteeing the behavior (e.g. any test that inserts and expects to find the document). `/BulkOperation/split` may be more likely to result in a test failure due to the large write (100,000 documents) taking longer to replicate. 

I expect the cause of the original error is the count operation runs on a different server which has not fully replicated the inserts. Alternatively: this PR can unskip the test without changes and wait for a possible future failure. Inspecting the server logs on a task failure may help to confirm this diagnosis.

Reply:
I am fine with proposed changes as-is.